x/superfluid/keeper/stake_test.go

```diff
@@ -663,334 +659,253 @@ func (suite *KeeperTestSuite) TestSuperfluidUnbondLock() {
-// 				err := suite.App.SuperfluidKeeper.SuperfluidUndelegate(suite.Ctx, lock.Owner, lock.ID)
-// 				suite.Require().NoError(err)
-// 			}
-// 			unbondingDuration := suite.App.StakingKeeper.GetParams(suite.Ctx).UnbondingTime
+			// refresh intermediary account delegations
+			suite.NotPanics(func() {
+				suite.App.SuperfluidKeeper.RefreshIntermediaryDelegationAmounts(suite.Ctx)
+			})
 
-// 			// check intermediary account changes after unbonding operations
-// 			for _, intAccIndex := range tc.checkAccIndexes {
-// 				expAcc := intermediaryAccs[intAccIndex]
-// 				suite.Ctx = suite.Ctx.WithBlockTime(suite.Ctx.BlockTime().Add(unbondingDuration + time.Second))
-// 				suite.App.EndBlocker(suite.Ctx, abci.RequestEndBlock{Height: suite.Ctx.BlockHeight()})
+			// ensure post-superfluid delegations osmo supplywithoffset is the same while supply is not
+			postsupply := suite.App.BankKeeper.GetSupply(suite.Ctx, bondDenom)
+			postsupplyWithOffset := suite.App.BankKeeper.GetSupplyWithOffset(suite.Ctx, bondDenom)
+			suite.Require().False(postsupply.IsEqual(presupply), "presupply: %s   postsupply: %s", presupply, postsupply)
+			suite.Require().True(postsupplyWithOffset.IsEqual(presupplyWithOffset))
+
+			originMultiplier := sdk.NewDec(20)
+			targetDelegations := []sdk.Dec{}
+			targetAmounts := []sdk.Int{}
+			for index, intAccIndex := range tc.checkAccIndexes {
+				expAcc := intermediaryAccs[intAccIndex]
+				multiplier, ok := multiplierByDenom[expAcc.Denom]
+				if !ok {
+					multiplier = originMultiplier
+				}
 
-// 				unbonded := suite.App.BankKeeper.GetBalance(suite.Ctx, expAcc.GetAccAddress(), sdk.DefaultBondDenom)
-// 				suite.Require().True(unbonded.IsZero())
-// 			}
+				targetDelegation := intermediaryDels[index].Mul(multiplier).Quo(originMultiplier)
+				lpTokenAmount := sdk.NewInt(1000000)
+				decAmt := multiplier.Mul(lpTokenAmount.ToDec())
+				asset := suite.App.SuperfluidKeeper.GetSuperfluidAsset(suite.Ctx, expAcc.Denom)
+				targetAmount := suite.App.SuperfluidKeeper.GetRiskAdjustedOsmoValue(suite.Ctx, asset, decAmt.RoundInt())
+				targetDelegations = append(targetDelegations, targetDelegation)
+				targetAmounts = append(targetAmounts, targetAmount)
+			}
 
-// 			// refresh intermediary account delegations
-// 			suite.NotPanics(func() {
-// 				suite.App.SuperfluidKeeper.RefreshIntermediaryDelegationAmounts(suite.Ctx)
-// 			})
-
-// 			// check changes after refresh operation
-// 			for _, intAccIndex := range tc.checkAccIndexes {
-// 				expAcc := intermediaryAccs[intAccIndex]
-// 				// check unbonded amount is removed after refresh operation
-// 				refreshed := suite.App.BankKeeper.GetBalance(suite.Ctx, expAcc.GetAccAddress(), sdk.DefaultBondDenom)
-// 				suite.Require().True(refreshed.IsZero())
-// 			}
-// 		})
-// 	}
-// }
+			for index, intAccIndex := range tc.checkAccIndexes {
+				expAcc := intermediaryAccs[intAccIndex]
+				valAddr, err := sdk.ValAddressFromBech32(expAcc.ValAddr)
```

Comment:
Can we group two for loops together for code sanity? Something like this:
```
	for index, intAccIndex := range tc.checkAccIndexes {
				expAcc := intermediaryAccs[intAccIndex]
				multiplier, ok := multiplierByDenom[expAcc.Denom]
				if !ok {
					multiplier = originMultiplier
				}

				targetDelegation := intermediaryDels[index].Mul(multiplier).Quo(originMultiplier)
				lpTokenAmount := sdk.NewInt(1000000)
				decAmt := multiplier.Mul(lpTokenAmount.ToDec())
				asset := suite.App.SuperfluidKeeper.GetSuperfluidAsset(suite.Ctx, expAcc.Denom)
				targetAmount := suite.App.SuperfluidKeeper.GetRiskAdjustedOsmoValue(suite.Ctx, asset, decAmt.RoundInt())

				valAddr, err := sdk.ValAddressFromBech32(expAcc.ValAddr)
				suite.Require().NoError(err)

				delegation, found := suite.App.StakingKeeper.GetDelegation(suite.Ctx, expAcc.GetAccAddress(), valAddr)
				if targetAmount.IsPositive() {
					suite.Require().True(found)
					suite.Require().Equal(delegation.Shares, targetDelegation)
				} else {
					suite.Require().False(found)
				}
			}
```

Reply:
Merged two loops into one - thanks for finding this